internal/jobcontainers/path.go

```diff
@@ -0,0 +1,79 @@
+package jobcontainers
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+// Checks the sandbox volume and PATH env variable to find application name. If the path
+// is relative, E.g. path\to\binary, it will be searched for in the sandbox volume. If it
+// is solely an application name then the base directory of the sandbox volume and all directories
+// in PATH will be searched. Job containers can run anything on the host so C:\path\to\exe,
+// D:\path\to\exe etc. are just passed through as is.
+func findExecutable(path string, imagePath string) (string, error) {
+	// Absolute path, just return the path. User specified (hopefully)
+	// a path to an executable on the host. C:\path\to\binary.exe
+	if filepath.IsAbs(path) {
+		return path, nil
+	}
+	// Not an absolute path, if no path seperators search in image path and if this fails
+	// search in PATH. If both of these fail then we error out. Otherwise if there are path
+	// seperators treat this as the user is trying to run an executable from the image directory
+	// and append the directory to the path supplied.
+	// E.g. path\to\binary.exe ---> C:\path\to\sandbox\ + path\to\binary.exe.
+	if filepath.Base(path) == path {
+		// User specified just the application name E.g. name_of_binary or name_of_binary.exe
+		// Check payload directory first and if this fails check PATH.
+		if absPath, err := searchPath(imagePath, path); err == nil {
+			return absPath, nil
+		}
```

Comment:
I don't understand why we can't just do the same thing we do for relative path below?

Reply:
I had some reason for this but it's eluding me now....